share/completions/mpc.fish

```diff
@@ -0,0 +1,91 @@
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a ls -d "List all files/folders in directory"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a search -d "Search for substrings in song tags"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a search -d " Search with a filter expression"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a find -d "Exact search with a filter expression"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a findadd -d "Find and add results to current queue"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a list -d "Return a list of all tags of given tag type"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a stats -d "Display statistics about MPD"
+
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a mount -d "List all mounts"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a mount -d "Create a new mount"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a unmount -d "Remove a mount"
+
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a outputs -d "List all available outputs"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a disable -d "Disable output(s)"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a enable -d "Enable output(s)"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a toggleoutput -d "Change status for given output(s)"
+
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a channels -d "List channels that other clients have subscribed to"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a sendmessage -d "Send a message to specified channel"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a waitmessage -d "Wait for at least one message on specified channel"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a subscribe -d "Subscribe to specified channel and continuously receive messages"
+
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a idle -d "Wait until an event occurs"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a idleloop -d "Keep waiting and printing events as they occur"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a version -d "Report version of MPD"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a volume -d "Set volume"
+
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a update -d "Scan for updated files in music directory"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a rescan -d "Like update, but also rescans unmodified files"
+complete -c mpc -n "not __fish_seen_subcommand_from $subcommands" -a current -d "Show currently playing song"
+
+complete -c mpc -n "__fish_seen_subcommand_from add insert" -a "(mpc listall)"
```

Comment:
Since there can be a lot of songs I wonder if it makes sense to use something based on `mpc ls $x` here where `$x` is the dirname of `commandline -ct`. What do you prefer?

Reply:
Reading the man page again I noticed that `search` can be used to search for a *substring* in the filename, and now that looks more useful that either of `'ls` or 'listall`. Also a problem with `ls` I see is that the argument should be the complete dirname, capitalization and all. What do you think ?

Reply:
It depends on the use case: adding all songs from some directory/artist is easier with `ls`, whereas adding a specific song is easier with `listall`.

I think both are fine and we can merge this as is.

> I noticed that search can be used to search for a substring in the filename

Yeah, but fish already filters completions based on the current token so the difference between `listall` and `search` might not matter. `search` might be more efficient.

Reply:
Yeah you're right; `search` vs `listall` doesn't matter much because of the token filtering. So yeah merging as is would be best I think.

> `search` might be more efficient

It could be on very large libraries, but I have only about 500 files, so can't say for sure. Anyways I'll add a comment about `search` being a possible faster alternative to `listall` for larger libraries.